src/Symfony/Bundle/TwigBundle/Resources/config/twig.php

```diff
@@ -0,0 +1,147 @@
+use Symfony\Bridge\Twig\Extension\WorkflowExtension;
+use Symfony\Bridge\Twig\Extension\YamlExtension;
+use Symfony\Bridge\Twig\Translation\TwigExtractor;
+use Symfony\Bundle\TwigBundle\CacheWarmer\TemplateCacheWarmer;
+use Symfony\Bundle\TwigBundle\DependencyInjection\Configurator\EnvironmentConfigurator;
+use Symfony\Bundle\TwigBundle\TemplateIterator;
+use Twig\Environment;
+use Twig\Extension\DebugExtension;
+use Twig\Loader\ChainLoader;
+use Twig\Loader\FilesystemLoader;
+use Twig\Profiler\Profile;
+use Twig\RuntimeLoader\ContainerRuntimeLoader;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+
+    ->set('twig', Environment::class)
+    ->public()
+    ->args([ref('twig.loader'), abstract_arg('Twig options')])
+    ->call('addGlobal', ['app', ref('twig.app_variable')])
+    ->call('addRuntimeLoader', [ref('twig.runtime_loader')])
+    ->configurator([ref('twig.configurator.environment'), 'configure'])
+    ->tag('container.preload', ['class' => 'Twig\Cache\FilesystemCache'])
+    ->tag('container.preload', ['class' => 'Twig\Extension\CoreExtension'])
+    ->tag('container.preload', ['class' => 'Twig\Extension\EscaperExtension'])
+    ->tag('container.preload', ['class' => 'Twig\Extension\OptimizerExtension'])
+    ->tag('container.preload', ['class' => 'Twig\Extension\StagingExtension'])
+    ->tag('container.preload', ['class' => 'Twig\ExtensionSet'])
+    ->tag('container.preload', ['class' => 'Twig\Template'])
+    ->tag('container.preload', ['class' => 'Twig\TemplateWrapper'])
+
+    ->alias('Twig_Environment', 'twig')
+    ->alias(Environment::class, 'twig')
+
+    ->set('twig.app_variable', AppVariable::class)
+    ->call('setEnvironment', [param('kernel.environment')])
+    ->call('setDebug', [param('kernel.debug')])
+    ->call('setTokenStorage', [ref('security.token_storage')->ignoreOnInvalid()])
+    ->call('setRequestStack', [ref('request_stack')->ignoreOnInvalid()])
+
+    ->set('twig.template_iterator', TemplateIterator::class)
+    ->args([ref('kernel'), abstract_arg('Twig paths'), param('twig.default_path')])
+
+    ->set('twig.template_cache_warmer', TemplateCacheWarmer::class)
+    ->args([ref('Psr\Container\ContainerInterface'), ref('twig.template_iterator')])
+    ->tag('kernel.cache_warmer')
+    ->tag('container.service_subscriber', ['id' => 'twig'])
+
+    ->set('twig.loader.native_filesystem', FilesystemLoader::class)
+    ->args([[], param('kernel.project_dir')])
+    ->tag('twig.loader')
+
+    ->set('twig.loader.chain', ChainLoader::class)
+
+    ->set('twig.extension.profiler', ProfilerExtension::class)
+    ->args([ref('twig.profile'), ref('debug.stopwatch')->ignoreOnInvalid()])
+
+    ->set('twig.profile', Profile::class)
+
+    ->set('data_collector.twig', TwigDataCollector::class)
+    ->args([ref('twig.profile'), ref('twig')])
+    ->tag('data_collector', ['template' => '@WebProfiler/Collector/twig.html.twig', 'id' => 'twig', 'priority' => 257])
```

Comment:
An idea: we could create an interface for tags as classes. 

```
interface ContainerTagInterface
{
    public function getName(): string;
    public function getAttributes(): array;
}
```

then allow something like `->tag((new DataCollectorTag)->template('...')->id('twig')->priority(257)` which makes tags discovorable, autocomplete and support for types.

Reply:
For ref, discoverable tags as constants for usage in the PHP config was discussed a bit in https://github.com/symfony/symfony/pull/24377